refactor: setup project for TypeScript

lpatiny opened the pull request.

lpatiny committed:
refactor: setup project for TypeScript

lpatiny commented:
@targos I tried to follow the documentation but still have one problem:

```
● Validation Error:

  Preset ts-jest/presets/js-with-ts not found.

  Configuration Documentation:
  https://jestjs.io/docs/configuration
  ```

targos commented:
`ts-jest` is missing from your dependencies: https://github.com/cheminfo/generator-cheminfo/blob/main/ts-migration.md#configure-jest

lpatiny committed:
docs: add jsdoc

lpatiny committed:
fix: add missing ts-jest dependency

lpatiny committed:
chore: add prettier

lpatiny committed:
chore: add node 16 to CI

lpatiny committed:
docs: fix function jsdoc argument type

lpatiny committed:
fix: add missing typescript dependency

lpatiny committed:
feat!: no default export

When using this project you will need to use destructuring:
`import {isAnyArray} from 'is-any-array'`

lpatiny committed:
docs: update README for destructuring

lpatiny committed:
test: fix destructuring

targos commented:
Eslint 8 just went out this weekend, I'm not surprised that our plugins don't support it yet.
You should remove the three `eslint-plugin-...` dependencies, `codecov`, and `eslint-config-cheminfo`.

lpatiny committed:
chore: remove useless dependencies

lpatiny committed:
chore: remove cheminfo-build and run-all
